### ISSUE TITLE
Prompt to save untitled file before run/debug

### ISSUE THREAD

**Issue by digitarald** [microsoft/vscode#111850]
Follow up from #110330, a common issue reported from new users who are just getting into coding is forgetting to save files before running them. This is probably based upon experience with cloud-based text and code editing that provide auto-save and live-refresh for an iterative editing cycle.

A stop gap solution that doesn't require too much thought is to prompt the file-save dialog when the file that the user wants to run is not saved to disk yet. I expect this to not be too surprising and only affect the corner cases of users who accidentally try to run new untitled files.

Related, #111053 is a good precursor, as more new files will have a language mode set from the beginning and therefor a file extension set when saving.

**Comment by isidorn** [comment]
@digitarald thanks for the feedback. But currently we automtically save all files before you start running / debugging. I just checked and this works just fine for me.

If this does not work for you can you please provide reproducable steps? Thanks!

**Comment by digitarald** [comment]
This was a theme in user tests, but I should have spend some more time on root cause analysis.

The core of the scenario is creating an _untitled_ file, which isn't save. The run/debug actions though generally don't work well without a workspace (even less so with an untitled file open), so maybe this is the actual root cause.

**Comment by isidorn** [comment]
Ok, adding a prompt to save active untitiled file makes good sense.
Duplicate of https://github.com/microsoft/vscode/issues/19415

@bpasero is there a better way to save current untitled file appart from executing `workbench.action.files.save` (that was 3 years ago)?

**Comment by bpasero** [comment]
@isidorn if you only know the resource, you can also use `ITextFileService.save` and pass it the resource. Better is to use `IEditorService#save` passing in the editor to save.

**Comment by isidorn** [comment]
@bpasero thanks, however I just found the `includeUntitled` flag in the `editorService.saveAll` so I went with that and it seems to work just fine.
@digitarald you can try it out with vscode insiders once we release it end of week and let me know how it works for you. Thanks

**Comment by bpasero** [comment]
Just fyi, I wonder if this may be annoying for users that always carry around some untitled files for notes or just in general. I think it would make more sense to only save untitled files with code that belong to the debug session and not all. How to detect that is a different question though...

**Comment by isidorn** [comment]
Good point. Let me change it to only do so for the active file, that way it should be better.

**Comment by isidorn** [comment]
@digitarald would be cool if you can try this out with vscode insiders and let me know if this works as expected so we can add a verified label

**Comment by digitarald** [comment]
Thank you for the ping! Worked as expected for new unsaved file.

**Comment by isidorn** [comment]
@digitarald thanks! Also no need to remove the verification-needed label, we just add the `verified` and the bot gets happy :)
